### PR TITLE
Emit events batch after a configurable delay

### PR DESCRIPTION
**Pull request by romac**
Closes: #3331

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Review by seanchen1991 — APPROVED** [review]
LGTM

File: crates/relayer/src/util/stream.rs [hunk 1]
```diff
@@ -90,6 +92,78 @@ where
...
+    let input = input.timeout(timeout);
+
+    stream! {
+        let mut state: Option<State<A>> = None;
+
+        for await x in input {
+            match x {
+                // timeout
+                Err(e) => {
+                    tracing::warn!("{e}");
```
Comment (ljoss17) [inline crates/relayer/src/util/stream.rs:118]: Should we keep this as `warn!`? With smaller delays it might flood the logs

Comment (romac) [reply crates/relayer/src/util/stream.rs:118]: The timeout is only emitted at most once per block, but yeah we should downgrade it anyway to debug.

**Review by ljoss17 — APPROVED** [review]
Looks good to me

**Review by ancazamfir — APPROVED** [review]
Amazing work @romac !

**Pull request merged**